src/containers/TemplateBuilder/shared/Parameters.tsx

```diff
@@ -26,9 +26,14 @@ export const Parameters: React.FC<ParametersProps> = ({
   const [asGui, setAsGui] = useState(true)
   const [isActive, setIsActive] = useState(false)
 
-  const sortedParameters = Object.entries(parameters).sort(([key1], [key2]) =>
-    key1 > key2 ? -1 : key1 === key2 ? 0 : 1
-  )
+  useEffect(() => {
+    // Sort elements only when opening modal
+    const sortedParameters = Object.entries(parameters)
+      .sort(([key1], [key2]) => (key1 > key2 ? -1 : key1 === key2 ? 0 : 1))
+      .reduce((sortedParameters, [key, element]) => ({ ...sortedParameters, [key]: element }), {})
+    setParameters(sortedParameters)
+  }, [])
```

Comment:
Just wondering why we need to sort them at all? What's wrong with using the "natural" order?

Reply:
Yeah, cos it re-sorts when you re-open the modal, so the parameter you last added is no longer on the bottom of the list, which is kind of annoying.

Reply:
I think that can be sorted later... For now seems ok to me to sort the parameters... It helps with finding things when you have a longer list of elements

Reply:
When you're making a new element, you're going to add parameters in a particular order that makes sense to you, probably starting with the simple ones, then adding the more complicated or niche parameters. Why would you want that order to change? When you have a list of 10 or so parameters, you want them to appear _in the order you entered them_, not get re-alphabetized when you re-open the modal. That's why we did this issue in the first place, we even agreed that's how it would work. This feels like we haven't actually solved the problem, just made it happen less often.

Reply:
Just an example from my own usage -- when I'm using the Modify Record action, there's usually loads of parameters because of all the fields that get updated. I find it really tricky to make sense of what's what. So what I'd normally do is add "tableName" first, then "MatchField", "matchValue", and then (finally) all the individual fields that need to be written to. It would be super annoying if, next time I open that Action, the parameters have all been sorted alphabetically instead of the logical, helpful order I added them. Why would we want to do this?

Reply:
I just spent a few minutes playing round with this and what I hoped doesn't actually work either. We can't really have *any* control over the key order, as the database has its own way of handling it when it comes to JSONB fields.

It's basically an arbitrary order, but I'd argue that that's no better or worse than alphabetical -- they're both pretty useless.

If we *did* want to change the ordering, then we can just to a simple sort on the "parameters" prop as it's passed into the "Parameters" component, but doesn't seem necessary right now.

What would be good is if it would show them in the order they're listed in the "plugin.json" config file -- so we should remember to revisit this when we do the issue to pre-fill parameters from the required list.